numba/cuda/compiler.py

```diff
@@ -756,6 +756,12 @@ def _prepare_args(self, ty, val, stream, retr, kernelargs):
             devrec = wrap_arg(val).to_device(retr, stream)
             kernelargs.append(devrec)
 
+        elif isinstance(ty, (types.UniTuple, types.Tuple, types.NamedUniTuple,
+                             types.NamedTuple)):
```

Comment:
Can this just check against `types.BaseTuple`?

Reply:
Looks like it can :-)